### PR TITLE
Added policy to audit VMs missing Backup Config

### PR DESCRIPTION
**Pull request by HarvestingClouds**
Added files for the policy to audit VMs missing backup configurations

### CODE REVIEW

File: samples/Backup/audit-vms-missing-backup-configurations/azurepolicy.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        "equals": "Microsoft.Compute/virtualMachines"
+      },
+      "then": {
+        "effect": "auditIfNotExists",
+        "details": {
+          "type": "Microsoft.RecoveryServices/backupprotecteditems",
+          "existenceCondition": {
+            "field": "name",
+            "like": "*"
+          }
```
Comment (camillemarie) [inline samples/Backup/audit-vms-missing-backup-configurations/azurepolicy.json]: This existenceCondition is not needed

File: samples/Backup/audit-vms-missing-backup-configurations/azurepolicy.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    "description": "Audit the VMs which are missing backup configurations in the environment",
+    "policyRule": {
+      "if": {
+        "field": "type",
+        "equals": "Microsoft.Compute/virtualMachines"
+      },
+      "then": {
+        "effect": "auditIfNotExists",
+        "details": {
+          "type": "Microsoft.RecoveryServices/backupprotecteditems",
```
Comment (camillemarie) [inline samples/Backup/audit-vms-missing-backup-configurations/azurepolicy.json]: >backupprotecteditems [](start = 46, length = 20)

nit:backupProtectedItems

(naming convention is to use camelCase for resourceTypes)

**Comment by MattiasAng** [comment]
@camillemarie This policy seems to have stopped working recently, do you have any idea why?
The same policy worked before and found non-compliant resources. Was something changed in the backend api for Microsoft.RecoveryServices/backupprotecteditems?

**Comment by camillemarie** [comment]
> @camillemarie These kind of policies seems to have stopped working recently, do you have any idea why?
> The same policy worked before and found non-compliant resources. Was something changed in the backend api for Microsoft.RecoveryServices/backupprotecteditems?

This is an open PR which was never merged to this repo. I don't know if it ever worked.

Tested it in my subscription today, and I see errors like the following in the Activity Log:
`{"ErrorCode":"InvalidContractVersionError","Message":{"Language":"en-US","Value":"Azure backup service version and Microsoft Azure Recovery Services Agent do not match.\r\nPlease install the latest version of Microsoft Azure Recovery Services Agent from the Microsoft Download Center (http://go.microsoft.com/fwlink/p/?LinkId=229525). If the issue persists, please contact Microsoft support."},"Values":[]}`

You can reproduce this by making the same GET request that the policy engine does :
`https://management.azure.com/subscriptions/<your subscription id>/resourcegroups/<your resource group id>/providers/microsoft.compute/virtualmachines/<your virtual machine name>/providers/Microsoft.RecoveryServices/backupprotecteditems?api-version=2017-07-01-preview`

I'm not familiar with the backupProtectedItems resource type, but this doesn't seem like correct behavior for a GET request to any resource provider. You should open a ticket with the Backup / Recovery Services team for this, especially if it was working before.

**Pull request closed without merging**